ivy/functional/ivy/nest.py

```diff
@@ -825,7 +876,9 @@ def nested_map(
             return class_instance(**dict(zip(x._fields, ret_list)))
         else:
             return class_instance(ret_list)
-    elif list_check_fn(x, list):
+    elif list_check_fn(x, list) or isinstance(x, extra_nest_types):
```

Comment:
isinstance(nest, (tuple, list) + extra_nest_types)

Reply:
This one is using `list_check_fn` for checking list type, so isn't this ok?

Reply:
one more thing I realised, `extra_nest_types` can be `Optional[Union[type, Tuple[type]]]`. So it can just be a type and not a tuple of types, in which case doing `(tuple, list) + extra_nest_types` wont work